File: src/java/quarkus/src/main/java/com/benchmark/api/model/JsonItem.java
package com.benchmark.api.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.UUID;

public class JsonItem {

    @JsonProperty
    public Integer id;

    @JsonProperty
    public String name;

    @JsonProperty
    public String description;

    @JsonProperty
    public String timestamp;

    @JsonProperty
    public String random;

    public JsonItem() {}

    public JsonItem(Integer id) {
        this.id = id;
        this.name = "Item " + id;
        this.description = "This is item number " + id;
        this.timestamp = java.time.Instant.now().toString();
        this.random = "data-" + UUID.randomUUID().toString();
    }

    @Override
    public String toString() {
        return "JsonItem{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: src/kotlin/ktor/src/main/kotlin/com/benchmark/models/JsonItem.kt
package com.benchmark.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.Instant
import java.util.UUID

@Serializable
data class JsonItem(
    val id: Int,
    val name: String,
    val description: String,
    val timestamp: String,
    val random: String
) {
    companion object {
        fun create(id: Int): JsonItem {
            return JsonItem(
                id = id,
                name = "Item $id",
                description = "This is item number $id",
                timestamp = Instant.now().toString(),
                random = "data-${UUID.randomUUID()}"
            )
        }
    }
}
